src/main/java/world/inetum/realdolmen/inetumrealJobs/controllers/AuthenticationController.java

```diff
@@ -82,40 +67,13 @@
                     .body(new MessageResponse("Error: Username is already taken!"));
         }
 
-        if (userRepository.existsByEmail(signUpRequest.getEmail())) {
-            return ResponseEntity
-                    .badRequest()
-                    .body(new MessageResponse("Error: Email is already in use!"));
-        }
-
         // Create new user's account
-        User user = new User(signUpRequest.getEmail(),
-                encoder.encode(signUpRequest.getPassword()));
-
-        Set<String> strRoles = signUpRequest.getRole();
-        Set<Role> roles = new HashSet<>();
-
-        if (strRoles == null) {
-            Role userRole = roleRepository.findByName(ERole.ROLE_JOB_SEEKER)
-                    .orElseThrow(() -> new RuntimeException("Error: Role is not found."));
-            roles.add(userRole);
-        } else {
-            strRoles.forEach(role -> {
-                switch (role) {
-                    case "admin":
-                        Role recruiterRole = roleRepository.findByName(ERole.ROLE_RECRUITER)
-                                .orElseThrow(() -> new RuntimeException("Error: Role is not found."));
-                        roles.add(recruiterRole);
-                        break;
-                    default:
-                        Role jobSeekerRole = roleRepository.findByName(ERole.ROLE_JOB_SEEKER)
-                                .orElseThrow(() -> new RuntimeException("Error: Role is not found."));
-                        roles.add(jobSeekerRole);
-                }
-            });
-        }
+        User user = new User(signUpRequest.getUsername(),
+                encoder.encode(signUpRequest.getPassword()),signUpRequest.getGender(), signUpRequest.getFirstName(),
+                signUpRequest.getLastName(), signUpRequest.getDateOfBirth(), signUpRequest.getStreetName(), signUpRequest.getHouseNumber(),
+                signUpRequest.getBox(), signUpRequest.getCity(), signUpRequest.getPostalCode(), signUpRequest.getCountry(),
+                signUpRequest.getMobilePhone(), signUpRequest.getProfilePicture(), signUpRequest.getRole());
```

Comment:
Seems a bit dangerous, what if an admin role is added? Can I just register myself as an admin then?